membership-attribute-service/test/services/AttributesMakerTest.scala

```diff
@@ -1,165 +1,240 @@
+        MembershipNumber = None,
+        AdFree = None,
+        TTLTimestamp = referenceDateAsDynamoTimestamp
+      )
+
+      val expected = Some(
+        Attributes(
+          UserId = identityId,
           Tier = None,
           RecurringContributionPaymentPlan = Some("Monthly Contribution"),
           MembershipJoinDate = None,
-          DigitalSubscriptionExpiryDate = None,
-          MembershipNumber = None,
-          AdFree = None,
-          TTLTimestamp = referenceDateAsDynamoTimestamp
+          DigitalSubscriptionExpiryDate = None
         )
+      )
 
-        val expected = Some(
-          Attributes(
-            UserId = testId,
-            Tier = None,
-            RecurringContributionPaymentPlan = Some("Monthly Contribution"),
-            MembershipJoinDate = None,
-            DigitalSubscriptionExpiryDate = None
-          )
-        )
+      val attributes = AttributesMaker.zuoraAttributesWithAddedDynamoFields(Some(zuoraAttributes), Some(dynamoAttributes))
 
-        val attributes = AttributesMaker.zuoraAttributesWithAddedDynamoFields(Some(zuoraAttributes), Some(dynamoAttributes))
+      attributes === expected
+    }
 
-        attributes === expected
-      }
+    "still return Zuora attributes when the user is not in Dynamo" in {
+      val zuoraAttributes = ZuoraAttributes(
+        UserId = identityId,
+        Tier = None,
+        RecurringContributionPaymentPlan = Some("Monthly Contribution"),
+        MembershipJoinDate = None,
+        DigitalSubscriptionExpiryDate = None
+      )
 
-      "still return Zuora attributes when the user is not in Dynamo" in {
-        val zuoraAttributes = ZuoraAttributes(
-          UserId = testId,
+      val expected = Some(
+        Attributes(
+          UserId = identityId,
           Tier = None,
           RecurringContributionPaymentPlan = Some("Monthly Contribution"),
           MembershipJoinDate = None,
           DigitalSubscriptionExpiryDate = None
         )
+      )
 
-        val expected = Some(
-          Attributes(
-            UserId = testId,
-            Tier = None,
-            RecurringContributionPaymentPlan = Some("Monthly Contribution"),
-            MembershipJoinDate = None,
-            DigitalSubscriptionExpiryDate = None
-          )
-        )
+      val attributes = AttributesMaker.zuoraAttributesWithAddedDynamoFields(Some(zuoraAttributes), None)
 
-        val attributes = AttributesMaker.zuoraAttributesWithAddedDynamoFields(Some(zuoraAttributes), None)
+      attributes === expected
+    }
 
-        attributes === expected
-      }
+    "return none if both Dynamo and Zuora attributes are none" in {
+      AttributesMaker.zuoraAttributesWithAddedDynamoFields(None, None) === None
+    }
 
-      "return none if both Dynamo and Zuora attributes are none" in {
-        AttributesMaker.zuoraAttributesWithAddedDynamoFields(None, None) === None
-      }
+  }
+
+  "actionAvailableFor" should {
```

Comment:
I think we're missing a case: "return false for a member who pays via PayPal"?

Reply:
Also, for my own understanding more than anything else, does the client decide which subset of paid users (members/subscribers/contributors) to show this message to?

Reply:
I am currently only setting actionAvailableFor for membership. We are already getting current subs and checking if there is a membership sub in 'AttributesMaker' so the filter happens [there](https://github.com/guardian/members-data-api/blob/2107776792c39231cd95fc4eb61842d612cb1eec/membership-attribute-service/app/services/AttributesMaker.scala#L49)

Reply:
Thanks for the explanation 👍

Reply:
I will add 'return false for a member who pays via PayPal'. Thanks for spotting that.